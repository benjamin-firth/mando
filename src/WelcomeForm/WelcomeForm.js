import React, { Component } from 'react';

class WelcomeForm extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return(
      <form>
        <label>Name</label>
        <input placeHolder='Galactic Identifier'></input>
        <label>Quote</label>
        <input placeHolder='Do or do not, there is no try.'></input>
        <label>Experience Level</label>
        <select>
          <option value='padawan'>Padawan</option>
          <option value='jediknight'>Jedi Knight</option>
          <option value='jedimaster'>Jedi Master</option>
        </select>
        <button>Enter, you must!</button>
      </form>
    )
  }
}

export default WelcomeForm;