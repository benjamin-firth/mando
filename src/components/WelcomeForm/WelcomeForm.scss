@import '../App/App.scss';

.visibleError {
  font-family: $main-font;
  color: RED;
}

.invisibleError {
  visibility: hidden;
}