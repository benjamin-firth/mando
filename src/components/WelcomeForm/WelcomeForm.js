import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './WelcomeForm.scss';


class WelcomeForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentName: '',
      currentQuote: '',
      currentRank: 'padawan',
      errorMsg: 'invisibleError'
    };
  }

  handleChange = (key, value) => {
    this.setState({[key]: value});
  }

  errorCheck = (e) => {
    e.preventDefault()
    const { currentName, currentQuote, currentRank } = this.state;
    if (currentName === '' || currentQuote === '') {
      this.setState({errorMsg: 'visibleError'});
    } else {
      this.setState({errorMsg: 'invisibleError'})
      this.props.history.push('/movie-page')
    }
  }

  render() {
    console.log(this.props.history);
    return(
      <form className='main login'>
        <label>Name</label>
        <input
          placeholder='Galactic Identifier'
          name='currentName'
          onChange={(e) => this.handleChange(e.target.name, e.target.value)}>
        </input>
        <label>Quote</label>
        <input
          placeholder='Do or do not, there is no try.'
          name='currentQuote'
          onChange={(e) => this.handleChange(e.target.name, e.target.value)}>
        </input>
        <label>Experience Level</label>
        <select
          name='currentRank'
          onChange={(e) => this.handleChange(e.target.name, e.target.value)}>
          <option value='padawan'>Padawan</option>
          <option value='jediknight'>Jedi Knight</option>
          <option value='jedimaster'>Jedi Master</option>
        </select>
        <button onClick={this.errorCheck} className='login-button' >Enter, you will!</button>
        <p className={this.state.errorMsg}>Enter all forms correctly, you must!</p>
        <img alt='Yoda shows up when login is invalid' src='https://1.bp.blogspot.com/-2zleuWnMiP4/XdmjP70dUPI/AAAAAAAAAZw/ZXK_NaTLn-cvCP7VIEHxNsA7zcaqEt6nACLcBGAsYHQ/s1600/baby-yoda-by-leo-batic.jpg' className={this.state.errorMsg} />
      </form>
    )
  }
}

export default WelcomeForm;
