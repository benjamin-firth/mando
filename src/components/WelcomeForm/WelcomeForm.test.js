import React from 'react';
import { shallow } from 'enzyme';
import WelcomeForm from './WelcomeForm.js'

describe('Welcome Form', () => {
  let wrapper;
  let mockChangePage;
  let mockErrorCheck;
  
  beforeEach(() => {
    mockChangePage = jest.fn();
    mockErrorCheck = jest.fn();
    wrapper = shallow(<WelcomeForm changePage={mockChangePage}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when handleChange is called', () => {
    const mockState = {
      currentName: '',
      currentQuote: '',
      currentRank: 'padawan',
      errorMsg: 'invisibleError'
    };
    const expected = {
      currentName: '',
      currentQuote: '',
      currentRank: 'Sith Lord',
      errorMsg: 'invisibleError'
    }
    wrapper.setState(mockState);
    wrapper.instance().handleChange('currentRank', 'Sith Lord');

    expect(wrapper.state()).toEqual(expected);
  })

  it('should call the errorCheck function when the button is clicked', () => {

    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().mockErrorCheck = jest.fn();

    wrapper.find('button').simulate('click', mockEvent);
    expect(wrapper.instance().mockErrorCheck).toHaveBeenCalled();
  })

  it('should run handleChange when the input is changed', () => {
    wrapper.instance().handleChange = jest.fn();
    const mockEvent = { 
      preventDefault: jest.fn(),
      target: {
        name: 'name',
        value: 'John Adams'
      }
    };

    wrapper.find('input').at(0).simulate('change', mockEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalled();

  })

  it('should run handleChange when the second input is changed', () => {
    wrapper.instance().handleChange = jest.fn();
    const mockEvent = { 
      preventDefault: jest.fn(),
      target: {
        name: 'quote',
        value: 'John Adams'
      }
    };

    wrapper.find('input').at(1).simulate('change', mockEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalled();

  })

  it('should run handleChange when the select is changed', () => {
    wrapper.instance().handleChange = jest.fn();
    const mockEvent = { 
      preventDefault: jest.fn(),
      target: {
        name: 'currentRank',
        value: 'Ultimate Jedi'
      }
    };

    wrapper.find('select').simulate('change', mockEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  })


})