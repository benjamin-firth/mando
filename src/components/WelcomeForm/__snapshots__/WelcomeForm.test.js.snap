// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Form should match the snapshot 1`] = `
<form
  className="main login"
>
  <label>
    Name
  </label>
  <input
    name="currentName"
    onChange={[Function]}
    placeholder="Galactic Identifier"
  />
  <label>
    Quote
  </label>
  <input
    name="currentQuote"
    onChange={[Function]}
    placeholder="Do or do not, there is no try."
  />
  <label>
    Experience Level
  </label>
  <select
    name="currentRank"
    onChange={[Function]}
  >
    <option
      value="padawan"
    >
      Padawan
    </option>
    <option
      value="jediknight"
    >
      Jedi Knight
    </option>
    <option
      value="jedimaster"
    >
      Jedi Master
    </option>
  </select>
  <button
    onClick={[Function]}
    type="button"
  >
    Enter, you will!
  </button>
  <p
    className="invisibleError"
  >
    Enter all forms correctly, you must!
  </p>
  <img
    alt="Yoda shows up when login is invalid"
    className="invisibleError"
    src="https://1.bp.blogspot.com/-2zleuWnMiP4/XdmjP70dUPI/AAAAAAAAAZw/ZXK_NaTLn-cvCP7VIEHxNsA7zcaqEt6nACLcBGAsYHQ/s1600/baby-yoda-by-leo-batic.jpg"
  />
</form>
`;
